### ISSUE TITLE
508-defect-2: List loops lack heading structure and unique labels 

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#45566]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## Details

Currently, the list loop patterns in CoE (and Burials) lack unique headings and labels
- Some list loop patterns allow users to add multiple generic `Add new [name of card e.g. service periods]` cards. This can create duplicative headings and fields as it will remain generic until added.
- _After_ adding multiple cards, they can be opened up simultaneously at the same time which creates multiple input fields with the same labels.
- In their non edited states, each card shares the same buttons for editing and deleting 

Ideally, the following could solve this problem (albeit as a crude bandaid):
- Only one new card can be added at any given time (solves the first issue)
- Only one card can be edited at any given time (solves the second issue) 
- Edit and delete buttons have `aria-labels` (e.g. Edit Marine Corps) based on the first heading within each card (which depends on the teams it seems like)

However, this gets a bit more complicated as I believe:
- The form system team is the only team capable of addressing the first two points 
- We're responsible for the heading structure of the cards

Ultimately, I recommend waiting to see how @thehastingsp and @Mottie 's new list loop pattern performs and extend that into CoE as well instead of trying to find a remediated solution immediately.

### Bandaid solution
- [ ] All cards have unique headings in a logical order
- [ ] Those unique headings are then used to uniquely label the nested inputs using a `legend` or `aria-describedby`

### Long term, recommended solution
- [ ] The new list loop alternative is implemented to replace all instances of the list loop within CoE and Burials

**Comment by saderagsdale** [comment]
@Mottie This was in blocked. Moving it to the product backlog.

**Comment by zack** [comment]
@saderagsdale @Mottie @MarciMcGuire 

I see the experimental design issue in the design system repo that was linked out from this ticket has been closed. Is there an updated status on that? Will the issues from this ticket be addressed by the design system? I'm not clear why this was moved out of "blocked" without the desired followup from FSR.

**Comment by saderagsdale** [comment]
@zack Please ask @steele-lm. We do not have current info about this ticket.

**Comment by steele-lm** [comment]
@zack The last comment on the related [design system component ticket](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1586) indicates the experimental design was added to the design system. Can we leverage the new component to fix this issue?

**Comment by zack** [comment]
@steele-lm it appears to still be a "candidate" in the design system, although I'm not sure exactly what that means in terms of future expectations or timelines. The problem is that while it's in the design system, it doesn't appear to be adopted into the _forms_ system yet. And this page is using the standard forms system. The solution here should be for the forms system to adopt this component, not for us to rebuild our form to _not_ use the form system.

I don't think this should be considered an "us" problem or a "CoE" problem, this is a forms problem. We just happen to be using the forms system (which we should be doing).

**Comment by steele-lm** [comment]
Thanks @zack! I'll follow up with the Platform Team to see if I can find out more.

**Comment by Mottie** [comment]
Sorry for not responding sooner. I don't know anything about the FSR app.

**Comment by steele-lm** [comment]
Form library is currently being developed by the Veteran-facing Forms team. It's not on their roadmap to bring in this component or Card specifically, but PRs are welcome.The challenge with this card component is that Design System Council didn't come to agreement on that Delete Button - Icon sub-component. Michael Dingee is working with Ryan to select a card design that could be tested.

**Comment by jeana-adhoc** [comment]
I'm currently reviewing tickets labeled with list-loop; is this still an issue in your form? Or can you share a link to the form that uses this pattern and I can take alook.
